node/loop.go

```diff
@@ -313,8 +327,13 @@ func (t *loopingTransport) loop() {
 		select {
 		case <-ctx.Done():
 			log.Printf("[DEBUG] Context done, setting deadlines to now")
+			t.readMu.Lock()
```

Comment:
HMmmm, I _think_ SetReadDeadline/SetWriteDeadline should at least be threadsafe, since that's just passing it on to the underlying net.Conn. Though I totally get the feeling of not wanting to risk anything anymore. :)

Reply:
I thought so too, but the gorilla go docs say it's not:

https://godoc.org/github.com/gorilla/websocket#hdr-Concurrency

> Applications are responsible for ensuring that no more than one goroutine calls the write methods (NextWriter, SetWriteDeadline, WriteMessage, WriteJSON, EnableWriteCompression, SetCompressionLevel) concurrently and that no more than one goroutine calls the read methods (NextReader, SetReadDeadline, ReadMessage, ReadJSON, SetPongHandler, SetPingHandler) concurrently.